lua/themepark/plugins/bbox.lua

```diff
@@ -0,0 +1,282 @@
+            if layer_config then
+                table.insert(layers, layer_config)
+            end
+        end
+    end
+    for _, data in ipairs(extra_layers) do
+        if not data.geometry_field then
+            data.geometry_field = 'geom'
+        end
+        if not data.srid then
+            data.srid = plugin.themepark.options.srid
+        end
+        table.insert(layers, data)
+    end
+    return layers
+end
+
+local function ordered_keys(o)
+    local keys1 = {}
+    local keys2 = {}
+    for k, v in pairs(o) do
+        if type(v) == 'table' then
+            table.insert(keys2, k)
+        else
+            table.insert(keys1, k)
+        end
+    end
+    table.sort(keys1)
+    table.sort(keys2)
+    for _,v in ipairs(keys2) do
+        table.insert(keys1, v)
+    end
+    return keys1
+end
+
+local function dump_toml(o, indent_size, parents)
+    indent_size = indent_size or 2
+    parents = parents or {}
+    if type(o) == 'table' then
+        local s = ''
+        local indent = string.rep(" ", #parents * indent_size)
+        -- sort keys for stable output order
+        for _, k in ipairs(ordered_keys(o)) do
+            local v = o[k]
+            if type(v) == 'table' then
+                local array_val = v[1] ~= nil
+                if array_val then
+                    table.insert(parents, k)
+                    s = s .. dump_toml(v, indent_size, parents)
+                    table.remove(parents)
+                elseif type(k) == 'number' then
+                    local tag = table.concat(parents, ".")
+                    s = s .. '\n\n' .. indent .. '[[' .. tag ..']]' .. dump_toml(v, indent_size, parents)
+                else
+                    table.insert(parents, k)
+                    indent = string.rep(" ", #parents * indent_size)
+                    local tag = table.concat(parents, ".")
+                    s = s .. '\n\n' .. indent .. '[' .. tag ..']' .. dump_toml(v, indent_size, parents)
+                    table.remove(parents)
+                end
+            else
+                if type(v) == 'string' then
+                    if string.find(v, '"') then
+                        v = '"""'..v..'"""'
+                    else
+                        v = '"'..v..'"'
+                    end
+                end
+                s = s .. '\n' .. indent .. k ..' = ' .. dump_toml(v, indent_size, parents)
+            end
+        end
+        return s
+    else
+        return tostring(o)
+    end
+end
+
+-- ---------------------------------------------------------------------------
+-- ---------------------------------------------------------------------------
+function plugin:write_config(filename, options)
+    if not options then
+        options = {}
+    end
+
+    local config = {
+        webserver = { server_addr = '0.0.0.0:8080' },
+        datasource = {
+            {
+                name = 'db',
+                postgis = {
+                    url = "postgres:///db"
```

Comment:
Is this hardcoding a database name? Not sure how this works with bbox, but maybe leave this empty so it picks up database connection from PG* env vars if they are available?

Reply:
BBOX currently requires a configuraton, but it can be overwritten with env vars. A possbile way is to add comments at the begin of the config file explaining this.

Is it possible to get the used db parameters for adding a matching PG url?

Reply:
The libpq library from PostgreSQL looks at some PG* env variables, this should work more or less everwhere unless that library is not used. So having no database config options is totally okay for osm2pgsql and should be for bbox also.

So there are several different ways of specifiying which database to use to osm2pgsql, maybe we can expose this information to Lua. But should we expose the original command line option or what's used in the end (taking env vars into account)? I have to think about this some more.